db/prefixeddb/prefixeddb.go

```diff
@@ -0,0 +1,124 @@
+package prefixeddb
+
+import (
+	"bytes"
+
+	"go.vocdoni.io/dvote/db"
+)
+
+// PrefixedDatabase wraps a db.Database prefixing all keys with `prefix`.
+type PrefixedDatabase struct {
+	prefix []byte
+	db     db.Database
+}
+
+// check that PrefixedDatabase implements the db.Database interface
+var _ db.Database = (*PrefixedDatabase)(nil)
+
+// NewPrefixedDatabase creates a new PrefixedDatabase.  If the db is already a
+// PrefixedDatabase, instead of wrapping again, the prefixes are appended to
+// avoid unecessay layers.
+func NewPrefixedDatabase(db db.Database, prefix []byte) *PrefixedDatabase {
+	if pdb, ok := db.(*PrefixedDatabase); ok {
+		return &PrefixedDatabase{append(pdb.prefix, prefix...), pdb.db}
+	}
+	return &PrefixedDatabase{prefix, db}
+}
+
+// Notice that this method doesn't close the wrapped db.Database.
```

Comment:
I wasn't too sure about what to do here.  My original reasoning was that the proper usage should be to always close from the parent Database, because you may close the wrapped Database thinking that the original is still opened, and then you're unable to operate on the original one.  But on the other hand, for Commit and Discard I do the action on the wrapped Tx.  So I'll just apply your suggestion.

Reply:
Yup, as you say, either all these methods should apply to the parent, or none of them should.